akka-http/src/main/scala/akka/http/scaladsl/unmarshalling/Unmarshaller.scala

```diff
@@ -141,6 +142,15 @@ object Unmarshaller
   final case class UnsupportedContentTypeException(supported: Set[ContentTypeRange])
     extends RuntimeException(supported.mkString("Unsupported Content-Type, supported: ", ", ", ""))
 
+  /** Order of parameters (`right` first, `left` second) is intentional, since that's the order we evaluate them in. */
+  final case class EitherUnmarshallingException(
+    rightClass: Class[_], right: Throwable,
+    leftClass: Class[_], left: Throwable)
+    extends RuntimeException(
+      s"Failed to unmarshal Either[${Logging.simpleName(leftClass)}, ${Logging.simpleName(rightClass)}] (attempted ${Logging.simpleName(rightClass)} first). " +
+        s"Right failure: ${right.getMessage}, " +
+        s"Left failure: ${left.getMessage}")
```

Comment:
No prob. One more argument would be that the happy (right) state should me more frequent than the exceptional (left) one so trying right first should be more efficient as it has a higher probability of success. Just my 2 cents.